### PR TITLE
Enable Link Time Optimization for Release builds

### PR DESCRIPTION
**Pull request by jonesmz**

### CODE REVIEW

**Comment by jhasse** [comment]
Thanks! Can you use lower case function names though?

**Comment by jonesmz** [comment]
> Thanks! Can you use lower case function names though?

Done.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,19 @@
 cmake_minimum_required(VERSION 3.12)
 project(ninja)
 
+if(CMAKE_BUILD_TYPE MATCHES "Release")
+  cmake_policy(SET CMP0069 NEW)
+  include(CheckIPOSupported)
+  check_ipo_supported(RESULT lto_supported OUTPUT error)
+
+  if( lto_supported )
```
Comment (jhasse) [inline CMakeLists.txt]: Remove the spaces

Comment (jonesmz) [reply CMakeLists.txt]: Done

**Comment by jonesmz** [comment]
> Can you use tabs for indention? (maybe we should convert the file to use 2 spaces instead though)

Done

**Comment by jhasse** [comment]
You have used 4 spaces now (or does GitHub render this incorrectly?).

**Comment by jonesmz** [comment]
> You have used 4 spaces now (or does GitHub render this incorrectly?).

I used the "tab" key...

Might be an editor "auto fix" issue.

I'll double check it.

**Comment by jonesmz** [comment]
It was an editor being "helpful".

Current version is tabs, as you asked for.

Apologizes for the trouble.

**Comment by jhasse** [comment]
No worries.

Binaries are smaller now, too! Thanks :)

**Pull request merged**